internal/server/httpapi/trigger.go

```diff
@@ -0,0 +1,398 @@
+
+		if variablesJSONRaw != "" {
+			if err := json.Unmarshal([]byte(variablesJSONRaw), &variables); err != nil {
+				http.Error(w,
+					fmt.Sprintf("failed to decode 'variables' json request param into a map: %s", err),
+					http.StatusInternalServerError)
+				return
+			}
+
+			for name, value := range variables {
+				v := &pb.Variable{
+					Name:   name,
+					Source: &pb.Variable_Cli{Cli: &empty.Empty{}},
+				}
+
+				if valBool, err := strconv.ParseBool(value); err == nil {
+					v.Value = &pb.Variable_Bool{Bool: valBool}
+				} else if valInt, err := strconv.ParseInt(value, 10, 64); err == nil {
+					v.Value = &pb.Variable_Num{Num: valInt}
+				} else {
+					// NOTE: for this case, it can either be a "string" or
+					// complex HCL type like an array or map. We can set this value
+					// as a Variable_Str here, and when we go to parse the variables
+					// later we do the proper string versus HCL check in variables.go
+					v.Value = &pb.Variable_Str{Str: value}
+				}
+
+				variableOverrides = append(variableOverrides, v)
+			}
+		}
+
+		var resp *pb.RunTriggerResponse
+		runTriggerReq := &pb.RunTriggerRequest{
+			Ref: &pb.Ref_Trigger{
+				Id: runTriggerId,
+			},
+			VariableOverrides: variableOverrides,
+		}
+
+		// attempt to make a grpc request to run trigger by id
+		if requireAuth {
+			resp, err = client.RunTrigger(ctx, runTriggerReq)
+		} else {
+			resp, err = client.NoAuthRunTrigger(ctx, runTriggerReq)
+		}
+		if err != nil {
+			log.Error("server failed to run trigger", "id", runTriggerId, "err", err)
+
+			if status.Code(err) == codes.PermissionDenied {
+				http.Error(w,
+					fmt.Sprintf("request not authorized to run trigger: %s", err),
+					http.StatusUnauthorized)
+			} else {
+				// improve http error code, which is more applicable for general queue failures?
+				http.Error(w,
+					fmt.Sprintf("server failed to run trigger: %s", err),
+					http.StatusPreconditionFailed)
+			}
+
+			return
+		}
+		if resp == nil {
+			http.Error(w,
+				fmt.Sprintf("server returned no job ids from run trigger %q", runTriggerId),
+				http.StatusInternalServerError)
+			return
+		}
+		jobIds := resp.JobIds
+
+		streamOutput := r.URL.Query().Get("stream")
+
+		// Attempt to stream output back, on request.
+		if streamOutput != "" {
+			if !requireAuth {
+				// We do not allow streaming job stream info if a no-auth token trigger was requested
+				log.Trace("server does not allow for streaming job stream output for no-token trigger URLs")
+				w.WriteHeader(http.StatusNoContent)
+				return
+			}
+
+			log.Debug("attempting to stream back queued job output from running trigger")
+
+			cn, ok := w.(http.CloseNotifier)
+			if !ok {
+				log.Error("failed to stream job output, could not create http.CloseNotifier")
+				http.NotFound(w, r)
+				return
+			}
+			flusher, ok := w.(http.Flusher)
+			if !ok {
+				log.Error("failed to stream job output, could not create http.Flusher")
+				http.NotFound(w, r)
+				return
+			}
```

Comment:
Yep, if the requester isn't capable I believe. And yeah, I don't know if `NotFound` is the right response here, probably more of an InternalServerError?

Reply:
Updated to be an InternalServerError instead, this makes more sense to me 👍🏻

Reply:
I would think a `400`  `StatusBadRequest`? It's not the servers fault I think? That's based on my assumption that the client made a request that it failed to indicate it can actually handle the response to. 

Not a big deal to worry about though, we can leave as-is if you would like